Rule/AdyenApi/UsedFallbackConfigRule.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AdyenPayment\Rule\AdyenApi;
+
+use Doctrine\Common\Persistence\ObjectRepository;
+
+final class UsedFallbackConfigRule implements UsedFallbackConfigRuleInterface
+{
+    /**
+     * @var ObjectRepository
+     */
+    private $shopRepository;
+    /**
+     * @var MainShopConfigRule
+     */
+    private $mainShopConfigRuleChain;
+
+    public function __construct(ObjectRepository $shopRepository, MainShopConfigRule $mainShopConfigRule)
+    {
+        $this->shopRepository = $shopRepository;
+        $this->mainShopConfigRuleChain = $mainShopConfigRule;
+    }
+
+    public function __invoke(int $shopId): bool
+    {
+        if (1 === $shopId) {
+            return false;
+        }
+
+        return ($this->mainShopConfigRuleChain)(
```

Comment:
Is `$this->mainShopConfigRuleChain` supposed to be a callable? Seems to return a boolean, or maybe I'm not understanding that class

Reply:
Under the namespace `AdyenPayment/Rule`
Are simple `Rules` that should only return booleans to indicate if the rule applies or not.

The rules are split up into smaller classes with SOLID programming in mind.  
That way they are easier to read / understand.  
Easier to test and less error prone.

Rules can be called directly or by using them in a chain, 
all depends where they are needed.

Mostly you want to combine rules, therefore the `Chain` class is used.

Benefit of the Chain class:
* one actual class 
* through Symfony DI, multiple unique chain's can be built through service configuration  
(no need to create new classes each time, only config needed)
* in return this offers more flexible way of code

Reply:
Nice setup about the rules and the rules chain. So in this line, `$this->mainShopConfigRuleChain` will return a boolean and then it will be && with `$this->shopRepository->...`? I guess syntax-wise I'm not following what it is trying to do in this return

Reply:
Got it now, __invoke is a magic method https://www.php.net/manual/en/language.oop5.magic.php#object.invoke